model/DeliveryFactory.php

```diff
@@ -230,4 +229,69 @@ protected function createDeliveryResource(
 
         return $compilationInstance;
     }
+
+    private function createNamespacedDeliveryResource(
+        KernelClass $deliveryClass,
+        array $additionalParameters
+    ): KernelResource {
+        $deliveryId = trim($additionalParameters[DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID] ?? '');
+
+        if (!$deliveryId) {
+            throw new RuntimeException(
+                sprintf('%s must not be empty.', DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID)
+            );
+        }
+
+        $this->getDeliveryDeleteService()
+            ->execute(
+                $this->createDeliverDeleteRequest($deliveryId)
+            );
+
+        $delivery = $deliveryClass->getResource($this->createNamespacedDeliveryId($deliveryId));
+
+        $delivery->setPropertiesValues([OntologyRdf::RDF_TYPE => $deliveryClass]);
+
+        $this->getEventManager()->trigger(new ResourceCreated($delivery));
+
+        return $delivery;
+    }
+
+    private function hasNamespace(): bool
+    {
+        return $this->hasOption(static::OPTION_NAMESPACE);
+    }
+
+    private function getNamespace(): string
+    {
+        $namespace = rtrim($this->getOption(static::OPTION_NAMESPACE, ''), '#');
+
+        if ($namespace === LOCAL_NAMESPACE) {
+            throw new DomainException(
+                "Overridden namespace value must be different from a local one, $namespace given"
+            );
+        }
+
+        return $namespace;
+    }
+
+    private function createNamespacedDeliveryId(string $deliveryId): string
+    {
+        return "{$this->getNamespace()}#$deliveryId";
```

Comment:
sprintf ,  as it would look more nice and easier to read

Reply:
Is it though?
I always found sprintf ugly as hell :P